src/StructuredLogger/Search/ProxyNode.cs

```diff
@@ -236,22 +236,17 @@ public string OriginalType
 
         private string GetProjectFileExtension()
         {
-            string result = null;
+            string result = "other";
 
-            if (Original is Project project)
+            if (Original is Project project && !string.IsNullOrEmpty(project.ProjectFileExtension))
             {
                 result = project.ProjectFileExtension;
             }
-            else if (Original is ProjectEvaluation evaluation)
+            else if (Original is ProjectEvaluation evaluation && !string.IsNullOrEmpty(evaluation.ProjectFileExtension))
             {
                 result = evaluation.ProjectFileExtension;
             }
 
-            if (result != null && result != ".sln" && result != ".csproj")

```

Comment:
This original code seems odd to single out csproj.  Is there a reason?

Reply:
no, it just forgets other project formats. I can't remember if we have a helper that stores all known project file extensions.